### PR TITLE
[Chore] Setup repository and CI

### PR DESCRIPTION
**Pull request by pasqu4le**
## Description

Initial files to set up the project and a CI for it.

## Related issue(s)

None.

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by pasqu4le** [comment]
Urgh, looks like the latest version of the libraries still hasn't been fetched by `hackage.nix`, they should be soon tho.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Run `make` to build everything.
+
+## For Contributors
+
+Please see [CONTRIBUTING.md](CONTRIBUTING.md) for more information.
+
+## About Serokell
+
+Homebase Lite is maintained and funded with ❤️ by [Serokell](https://serokell.io/).
+The names and logo for Serokell are trademark of Serokell OÜ.
```
Comment (lierdakil) [inline README.md:25]: I don't think this is quite right.

File: docs/branching.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<!--
+   - SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+   -
+   - SPDX-License-Identifier: CC0-1.0
```
Comment (lierdakil) [inline docs/branching.md:4]: Do we _want_ to keep Serokell copyrights here (and also in a bunch of other places)

**Review by lierdakil — COMMENTED** [review]
Pretty much our typical scaffolding, but I see a bunch of Serokell copyrights, including in README and CONTRIBUTING. I have to wonder if those are an accident.

File: LICENSES/LicenseRef-ReplaceMe.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+CC0-1.0.txt
```
[the remark this answers is not in the document]
Comment (pasqu4le) [reply LICENSES/LicenseRef-ReplaceMe.txt:1]: indeed, I suppose that `reuse` didn't flag it as unused due to your other comment :thinking:

File: README.md (hunk 1 above)
Comment (pasqu4le) [reply README.md:25]: It isn't at all, I thought I had removed this section in favor of the `License` one, but apparently I left it behind by mistake.

File: docs/branching.md (hunk 2 above)
Comment (pasqu4le) [reply docs/branching.md:4]: There are a few files, including this one, that are exact copies from elsewhere (mostly [metatemplates](https://github.com/serokell/metatemplates)), that aren't expected to change and shouldn't really be copywritable (see [our FAQ](https://www.notion.so/serokell/FAQ-8da3477b8b7c4faaab6460ff38fc8929) on the matter).

So, I checked again and changed some of them that were using our `CC0-1.0`, but I left the ones that fall under the category above.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  * Add the following header at the very top (but below the shebang, if there
+    is one) of each source file in the repository (yes, each and every source
+    file – it is not as hard as it might sound):
+
+    ```haskell
+    -- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+    --
+    -- SPDX-License-Identifier: LicenseRef-ReplaceMe
+    ```
```
[the remark this answers is not in the document]
Comment (pasqu4le) [reply CONTRIBUTING.md:52]: :facepalm: indeed

> do we actually want to use multiple licences in this here repo?

Well, in some place we need to preserve someone else's licence, see `LicenseRef-MIT-nmattia`, and for some files we want to keep the one that they already have from elsewhere (see comment below) .

For new files however we'll use the `MIT` for `Tezos Commons` everywhere tho.

**Comment by pasqu4le** [comment]
@lierdakil I addressed your comments and made a few more small changes, see the fixup commit.

**Review by lierdakil — APPROVED** [review]

**Pull request merged**